[Ian Yong] iP

ianyong opened the pull request.

ianyong committed:
Add Level 1: Greet, Echo, Exit (#9)

* Add PrettyPrinter

* Add REPL

ianyong committed:
Add Level 2: Add, List (#10)

* Add Task

* Add TaskManager

ianyong committed:
Add Level 3: Mark as Done (#11)

* Track completion state in Task

* Add 'markAsDone' method to TaskManager

* Catch exceptions when marking a task as done

ianyong committed:
Add Level 4: ToDos, Events, Deadlines (#12)

* Add ToDo

* Add Deadline and Event

* Refactor adding of tasks to be more generic

* Update output to match Level 4's example

* Use string resources instead of hard coded strings

* Extract remaining hard coded strings to the resource file

* Update Javadoc descriptions

ianyong committed:
Add A-Enums: Use Enums if Applicable (#13)

* Refactor classes into packages

For better organisation

* Add Command enum

ianyong committed:
Add Level 5: Handle Errors (#14)

* Add validation for the 'done' command

* Add validation for the 'deadline' and 'event' commands

* Add validation for the 'bye' and 'list' commands

* Add validation for the 'todo' command

* Make use of enum abstract method

* Improve tokenisation of user input

ianyong committed:
Add Level 6: Delete (#15)

* Add remove task functionality to TaskManager

* Add 'delete' command to REPL

ianyong committed:
Add A-TextUiTesting: Automated Text UI Testing (#16)

* Get runtest.bat working with packages and properties files

* Get runtest.sh working with packages and properties files

* Set output encoding for runtest.bat to UTF-8

* Add input and expected output

ianyong committed:
Add Level 7: Save (#26)

Squashed commit of the following:

commit d7dc437c659b5d724ed060c8530438566ddd26ae
Author: Ian Yong <ianyongyc@gmail.com>
Date:   Tue Aug 25 16:54:42 2020 +0800

    Fix NullPointerException when file does not exist

commit 58d56eecacaf3e80df19d87d47554629d4683e0a
Author: Ian Yong <ianyongyc@gmail.com>
Date:   Tue Aug 25 16:38:33 2020 +0800

    Add state retrieval to PersistentList

commit da7edf2508057a443c8a0d1cbb3f719c5ec68ee2
Author: Ian Yong <ianyongyc@gmail.com>
Date:   Tue Aug 25 14:00:40 2020 +0800

    Set encoding of System.out to UTF-8

commit 1414a9b6f1cde6d057e91c0ddb6818b8ff682af8
Author: Ian Yong <ianyongyc@gmail.com>
Date:   Tue Aug 25 13:50:31 2020 +0800

    Fully implement PersistentList

commit 8ed2c196218f279e0eeac42dd2fd22118df9c48f
Author: Ian Yong <ianyongyc@gmail.com>
Date:   Tue Aug 25 03:04:08 2020 +0800

    Fix error when file parent dir does not exist

commit 5d00998567b59cd1d18abad5f0038942cce85f29
Author: Ian Yong <ianyongyc@gmail.com>
Date:   Tue Aug 25 03:03:40 2020 +0800

    Update .gitignore

    Ignore /data

commit 18866e61d81631ac71fa2b96d9be068cbdac1c85
Author: Ian Yong <ianyongyc@gmail.com>
Date:   Tue Aug 25 02:59:39 2020 +0800

    Add PersistentList

commit 05791c1b024f77928c2e8b0afcb549b2d89d25ba
Author: Ian Yong <ianyongyc@gmail.com>
Date:   Tue Aug 25 02:13:00 2020 +0800

    Add StorageManager

commit 63b3ff88cab30d5f8ac98909a547d345e0281298
Author: damithc <damithch@damithch-mbp.comp.nus.edu.sg>
Date:   Mon May 25 00:58:18 2020 +0800

    Add Gradle support

ianyong committed:
Add Level 8: Dates and Times (#27)

Squashed commit of the following:

commit 654ce37d4bf74c7224223a5a1d70b28fb67579a1
Author: Ian Yong <ianyongyc@gmail.com>
Date:   Tue Aug 25 18:42:22 2020 +0800

    Add 'overdue' command to display overdue tasks

commit 4317ee522fcc6b5bf7c4508f43f0f23ddfe595be
Author: Ian Yong <ianyongyc@gmail.com>
Date:   Tue Aug 25 18:37:41 2020 +0800

    Add 'upcoming' command to display upcoming tasks

commit b552e10583529faeb11cc152c902c10ae497d568
Author: Ian Yong <ianyongyc@gmail.com>
Date:   Tue Aug 25 18:02:02 2020 +0800

    Format displayed datetime

commit 61cedad408d8bff1fb67e0f899129cf97dfaaafd
Author: Ian Yong <ianyongyc@gmail.com>
Date:   Tue Aug 25 17:48:16 2020 +0800

    Update Deadline and Event tasks to use LocalDateTime

commit e0c9eb430864db278dbc5248cfdd5029001a242e
Author: Ian Yong <ianyongyc@gmail.com>
Date:   Tue Aug 25 17:27:45 2020 +0800

    Add DateTimeParser

ianyong committed:
Fix serialization issues with LocalDateTime

Add LocalDateTimeAdapter to allow gson to serialize and de-serialize
LocalDateTime objects properly.

ianyong committed:
Add a top level package (#28)

ianyong committed:
Add JUnit tests for Deadline and Event (#29)

ianyong committed:
Update build.gradle to work with the Shadow plugin (#30)

ianyong committed:
Add A-CodingStandard: Follow the Coding Standard (#31)

Squashed commit of the following:

commit dba33eb0697729d0f3950b0a27b8d4357ff65915
Author: Ian Yong <ianyongyc@gmail.com>
Date:   Tue Aug 25 22:57:13 2020 +0800

    Update switch statement styling

    This is due to CS2103T having this inane requirement whereby the
    body of a switch statement is not to be indented.

ianyong committed:
Add Level 9: Find (#32)

Squashed commit of the following:

commit a4b5df944f252c551300a6cd6c57981fa3b116bb
Author: Ian Yong <ianyongyc@gmail.com>
Date:   Tue Aug 25 23:14:25 2020 +0800

    Exclude tasks which are done from 'overdue' and 'upcoming'

commit d636eb8d7e732747cd23ad7c6e6f5581a38dada7
Author: Ian Yong <ianyongyc@gmail.com>
Date:   Tue Aug 25 23:10:44 2020 +0800

    Add 'find' command

ianyong committed:
Add Checkstyle

ianyong committed:
Fix Checkstyle errors

TianYong-Goh commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,71 @@
+package duke.tasks;
+
+import duke.utils.ResourceHandler;
```
Perhaps you could consider putting a space between line 3 and 4 which follow the other classes' import's grouping?

TianYong-Goh commented on src/main/java/duke/tasks/ToDo.java:
```diff
@@ -0,0 +1,48 @@
+package duke.tasks;
+
+import duke.utils.ResourceHandler;
```
Perhaps you could consider putting a space between line 3 and 4 which follow the other classes' import's grouping?

TianYong-Goh commented on src/main/java/duke/tasks/TaskManager.java:
```diff
@@ -0,0 +1,133 @@
+    }
+
+    /**
+     * Mark a {@code Task} as done.
```
Consider changing Mark to Marks to follow Java coding Standard.

ianyong replied:
Thanks for catching this!

TianYong-Goh reviewed (commented):
Works do follow the Java coding standard very well. Hard to find anything that violates the rules.

TianYong-Goh commented on src/main/java/duke/Repl.java:
```diff
@@ -0,0 +1,123 @@
+    private static final Scanner scanner = new Scanner(System.in);
+    /** {@code PrettyPrinter} object for formatting the REPL output. */
+    private static final PrettyPrinter prettyPrinter = new PrettyPrinter(LEFT_PADDING_SIZE, DIVIDER_LENGTH);
+    /** {@code TaskManager} object to keep track of tasks */
```
It looks like you miss out a full stop behind the tasks.

TianYong-Goh commented on src/main/java/duke/enums/Command.java:
```diff
@@ -0,0 +1,236 @@
+        /**
+         * Validates whether the user input is of the correct format for the 'overdue' command.
+         *
+         * @param input the user input.
```
Perhaps you might want to consider putting Cap for the first letter of the description. For example: The user input

ianyong replied:
Hmm, the reason I didn't do so is because the CS2103T coding standard doesn't explicitly list this as necessary (although the example does capitalise the first letter of the parameter descriptions). I've been mostly following the Java standard library JavaDoc format which does not practise capitalisation.

TianYong-Goh replied:
Oooo ok. thanks for the explanation:)

TianYong-Goh commented on src/main/java/duke/utils/LocalDateTimeAdapter.java:
```diff
@@ -0,0 +1,63 @@
+     *
+     * @param src       the object that needs to be converted to Json.
+     * @param typeOfSrc the actual type (fully genericized version) of the source object.
+     * @param context
```
It looks like you accidentally miss out the description for context.

ianyong committed:
Add JavaFX

ianyong committed:
Refactor Repl

Split the logic which handles the processing of user input from
the I/O loop.

ianyong committed:
Move Schedulable interface to the tasks package

ianyong committed:
Add basic GUI

ianyong committed:
Add command line arguments

This is to allow the user to pass in a command line argument to
start Duke in CLI mode.

ianyong committed:
Update application exiting mechanism

ianyong committed:
Add dark mode

ianyong committed:
Make scrollbar dark

Also added title and disabled window resizing.

ianyong committed:
Update images and application icon

ianyong committed:
Add welcome message to GUI

Also fixed long texts being truncated.

ianyong committed:
Disallow empty inputs and colour user messages green

ianyong committed:
Modify 'find' to treat each argument as a separate keyword

ianyong committed:
Merge pull request #37 from ianyong/ian/add-A-checkstyle

Add A-CheckStyle: Use CheckStyle

ianyong committed:
Merge branch 'master' into branch-Level-10

ianyong committed:
Merge pull request #38 from ianyong/branch-Level-10

Add Level 10: GUI

ianyong committed:
Merge branch 'master' into ian/add-var-args

ianyong committed:
Merge pull request #39 from ianyong/ian/add-var-args

Add A-Varargs: Use Varargs

ianyong committed:
Fix styling issues (#40)

* Fix CheckStyle errors

* Fix issues highlighted in review

ianyong committed:
Add assertions

ianyong committed:
Refactor long switch statement into enum methods

ianyong committed:
Refactor 'Command.execute' to return a 'DukeResponse'

ianyong committed:
Use lambdas and streams instead of for loops

ianyong committed:
Merge pull request #47 from ianyong/branch-A-Assertions

Add A-Assertions: Use Assertions

ianyong committed:
Merge branch 'master' into branch-A-CodeQuality

ianyong committed:
Merge pull request #48 from ianyong/branch-A-CodeQuality

Add A-CodeQuality: Improve Code Quality

ianyong committed:
Merge pull request #49 from ianyong/branch-A-Lambdas-Streams

Add A-Lambdas: Use Lambdas & A-Streams: Use Streams

ianyong committed:
Set up Continuous Integration

ianyong committed:
Merge pull request #50 from ianyong/branch-A-CI

Add A-CI: Set up CI

ianyong committed:
Add PersistentMap

ianyong committed:
Add Store to hold the application state

ianyong committed:
Add AliasManager

ianyong committed:
Handle aliasing invalid commands

ianyong committed:
Fix crash on invalid command

ianyong committed:
Stop aliases from using the same name as existing commands

ianyong committed:
Merge pull request #51 from ianyong/branch-BCD-Extension

Add BCD-Extension: Add an extension

ianyong committed:
Add alias removal functionality (#52)

* Refactor command validation and execution to work with args only

* Use aliases when displaying error messages

* Add alias removal functionality

* Make AliasManager responses dynamic

ianyong committed:
Add rounded corners to display pictures (#55)

ianyong committed:
Add 'help' command (#56)

ianyong committed:
Add mandarin support (#57)

* Refactor ResourceHandler

* Add mandarin strings

* Update StorageManager to use UTF-8 encoding when writing

* Add ConfigManager

* Add 'language' command to switch between languages

ianyong committed:
Set theme jekyll-theme-tactile

ianyong committed:
Add user guide (#58)

* Add user guide

* Add Ui.png

ianyong committed:
Set theme jekyll-theme-modernist

ianyong committed:
Add command history (#59)

* Add LinkedListNode

* Add EvictingStack

* Add CommandHistory

* Add command history functionality

* Set caret position on command change

ianyong committed:
Make GUI resizable (#60)

* Make conversation history resizable

* Set minimum window size

* Add anchors to text field and button

* Fix mouse wheel scrolling not working

* Add mention of 'help' command on startup

ianyong committed:
Improve user guide (#61)

* Update user guide introduction

* Update Ui.png

* Add drop shadows to images

* Add user guide features

ianyong committed:
Fix GitHub Pages styling

ianyong committed:
Update README
